src/main/java/com/sst/utopia/booking/controller/BookingController.java

```diff
@@ -0,0 +1,153 @@
+package com.sst.utopia.booking.controller;
+
+import java.util.NoSuchElementException;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.sst.utopia.booking.model.PaymentAmount;
+import com.sst.utopia.booking.model.SeatLocation;
+import com.sst.utopia.booking.model.Ticket;
+import com.sst.utopia.booking.model.User;
+import com.sst.utopia.booking.service.BookingService;
+
+/**
+ * Controller to present the booking service to the microservices that provide
+ * the client-facing API.
+ *
+ * @author Jonathan Lovelace
+ */
+@RestController
+@RequestMapping("/booking")
+public class BookingController {
+	/**
+	 * Service class used to handle requests.
+	 */
+	@Autowired
+	private BookingService service;
+	/**
+	 * Reserve a ticket for the given seat.
+	 * FIXME: Allow getting the user from headers (injected by the security layer)
+	 * @param flight the flight number of the flight
+	 * @param row the row number of the seat
+	 * @param seat the seat within the row
+	 * @param user the user details
+	 */
+	@PostMapping("/book/{flight}/{row}/{seat}")
```

Comment:
Need to change this to `/book/flights/{flight}/rows/{rows}/seats/{seat}`, per Akbar.